### DISCUSSION TITLE
Resilient message sending

### DISCUSSION THREAD

**Discussion by ozrecsec** [jchristn/WatsonTcp#133] [General]
Hi, in my project I'm going to implement a sort of resilient message sending (like a queue).

I's intended only for "from client to server" messages and not "from server to client". It will work only for Send method, not SendAndWait, with an optional parameter (bool resilient).

In few words my client sends immediatly the message to the server if it is connected, otherwise it stores the message locally and sends it as soon the connection is estabilished.

for storage i think I'll use a json file for each message. In settings will be the folder for storage.

Do you think this is a interesting addon in Watson or it's better I do this in my wrapping library?

thanks

**Comment by jchristn** [comment]
Hi @ozrecsec thanks for your note.  The use case is pretty clear, if it's connected, send, and if not, store it somewhere, and when it reconnects, drain the queue.  This kind of functionality is great but if integrated into WatsonTcp would add a lot of weight.  There are a lot of packages out there that would help with the storage side (like PersistentDictionary).  Cheers, Joel

**Reply by ozrecsec** [reply]
ok, thanks, I'll do it in my library ;-)
